### DISCUSSION TITLE
Webserver Home V3 hangs from time to time

### DISCUSSION THREAD

**Discussion by GalOR7** [solectrus/solectrus#2541] [General]
Did you notice that the little webserver in the V3 hangs from time to time (every week). The info website is not reachable anymore in this situation and also Solectrus show the disconnected status. The only remedy is to reset/reboot the device. All the rest of the functionality seems fine. App and Senec Homepage still works. My Askoheat element for generating hot water seems also not getting feeding informations from the V3. Would be interesting for me to get your feedback. BR GalOR

**Comment by ledermann** [comment]
This seems to be the [famous](https://solectrus.de/blog/2023-09-06-erfahrungen-mit-dem-senec-update/) NPU error, [see](https://github.com/solectrus/senec-log-downloader) the logs. Most of the time the device is back online after a few minutes, but sometimes a reboot seems to be required. From my point of view, the connection to SENEC is interrupted during an NPU error, so neither the app nor mein-senec.de will receive current data.

**Reply by GalOR7** [reply]
Looks different here. I got an update for the NPU error a few weeks ago. Since this time I had no NPU errors any more. Neither display nor the logs. I remeber you wrote somewhere you made a new query methode of the home v3 in order to reduce the load of the device. I will review this carefully and let you know. It is not too anyoing as the app still works but a pity as I really like solcetrus real time dashboard :-)

**Reply by ledermann** [reply]
SENEC-Collector [v0.10.0](https://github.com/solectrus/senec-collector/releases/tag/v0.10.0) or higher uses persistent connections, so each request to the SENEC device takes about 8ms. I think this minimizes the load. 

Still, I see NPU errors in the log files about once a week (MCU 0826, NPU-Image 2310). This is annoying, but not a big deal.

**Reply by GalOR7** [reply]
Okay I will check this and let you know

**Reply by GalOR7** [reply]
Yes indeed with your log script is cool stuff. Have seen yesterday the 21st there was a NPU switch off. This the first case since 60days. Since then I got rid of the solectrus not connected messages. This is what I have with fw 2023-10-04 11:51:46 [I|Settings] MCU firmware version: 0826, GUI version 0966
This is what you explain in the BLOG so seems the most recent state in my v3
